client/data/deposits/selectors.js

```diff
@@ -46,7 +57,51 @@ export const getDepositsOverviewError = ( state ) => {
  */
 export const getAllDepositsOverviews = ( state ) => {
 	const DepositsOverview = getDepositsState( state ).overviews || {};
-	return DepositsOverview.data;
+
+	// Return an empty skeleton if data has not been loaded yet.
+	if ( ! DepositsOverview.data ) {
+		return {
+			account: null,
+			currencies: [],
+		};
+	}
+
+	const { deposit, balance, account } = DepositsOverview.data;
+	const groups = {
+		lastPaid: deposit.last_paid,
+		nextScheduled: deposit.next_scheduled,
+		pending: balance.pending,
+		available: balance.available,
+	};
+
+	const currencies = {};
+	for ( const [ key, values ] of Object.entries( groups ) ) {
+		values.forEach( ( value ) => {
+			const { currency } = value;
+
+			if ( ! currencies[ currency ] ) {
+				currencies[ currency ] = {
+					currency,
+					lastPaid: null,
+					nextScheduled: null,
+					pending: null,
+					available: null,
+				};
+			}
+
+			// There will be a single deposit/balance per currency, no arrays here.
+			currencies[ currency ][ key ] = value;
+		} );
+	}
+
+	return {
+		account,
+
+		// The default currency should appear at the top of the list.
+		currencies: Object.values( currencies ).sort( ( a, b ) => {
+			return account.default_currency === b.currency ? 1 : 0;
```

Comment:
Some interesting logic here, try this snippet to see how it works:

```js
[ 'eur', 'gbp', 'bgn', 'huf', 'bam' ].sort( ( a, b ) => ( 'gbp' === b ) ? 1 : 0 )
```

Ideally the default currency should be first, and the order of the rest can remain the same.

Reply:
Thank you for the suggestion, but it doesn't work either. Turns out that browsers treat this differently.

It works in Chrome:

```js
[ 'eur', 'gbp', 'bgn', 'huf', 'bam' ].sort( ( a, b ) => ( 'gbp' === b ) ? 0 : -1 )
(5) ["gbp", "bam", "huf", "bgn", "eur"]
```

But not in Firefox:

```js
[ 'eur', 'gbp', 'bgn', 'huf', 'bam' ].sort( ( a, b ) => ( 'gbp' === b ) ? 0 : -1 )
Array(5) [ "eur", "gbp", "bgn", "huf", "bam" ]
```

The way I initially implemented it was the other way around.

To avoid this, I'm just filtering the array twice and concatenating the result. Not as elegant, but definitely more reliable.